articles/react-pwa/index.md

```diff
@@ -0,0 +1,539 @@
+            <div className="todo">
+            <li className={todo.completed?"checked":""}>
+                {todo.text}
+            </li>
+            <button onClick={HandleComplete} className="check-btn">{todo.completed?"Uncheck":"Check"}</button>
+            <button className="delete-btn" onClick={HandleDelete}>Delete</button>
+            </div>
+        </div>
+    );
+}
+
+export default Todo;
+```
+
+### The App component 
+Finally, we need to include the App component. The app component was the main parent component of the `TodoList` and the `Form` components. I have included a CSS file for our styling named *App.css* which is in the same `src` root folder as our *App.js* file.
+
+```javascript
+import React,{useState} from 'react';
+
+// components
+import Form from './components/Form';
+import TodoList from './components/TodoList';
+
+// CSS files
+import './App.css'
+
+const App=()=>{
+
+  const [todos,setTodos]=useState([]);
+
+  const addTodo= todo=>{
+          const newTodos=[todo,...todos];
+           setTodos(newTodos);
+       }
+  
+  return (
+    <div>
+      <h1>TODO PWA</h1>
+      <Form addTodo={addTodo}/>
+      <TodoList todos={todos} setTodos={setTodos}/>
+    </div>
+  );
+}
+
+export default App;
+```
+
+The App component walkthrough:
+
+- We import the `TodoList` and `Form` component and the CSS file just below the `React` import.
+
+- Inside the function, the `useState` hook will create our state for todos. 
+
+- The `addTodo` is a function that is used to add a new todo calling `setTodos` method to update the state.
+
+- We use JSX to return a div containing `<h1>` tag as the header. The `Form` and `TodoList` components are mounted passing along the appropriate props.
+
+### Application Styling
+This tutorial assumes you know basics of CSS styling, so I have a basic CSS file: 
+
+```css
+body{
+    background-color: rgb(72, 72, 247);
+    font-size: 1.2rem;
+    max-width: 100vw;
+    max-height: 100vh;
+}
+
+input{
+    padding:1.3rem 5rem;
+    border:none;
+    font-size: 1.1rem;
+}
+.add-btn{
+    padding: 1.3rem;
+    font-size: 1.1rem;
+}
+
+button{
+    border:none;
+    margin-right: 5px;
+
+}
+
+button:hover{
+    transform: scale(0.98);
+    transition: linear 0.8s;
+    cursor: pointer;
+}
+.checked{
+    color:rgb(240, 240, 247);
+    text-decoration: line-through;
+}
+.todo{
+    display: flex;
+}
+
+.check-btn{
+    color:rgb(14, 14, 14);
+    background-color: rgb(95, 130, 245);
+}
+.delete-btn{
+    color: rgb(247, 238, 242);
+    background-color: rgb(255, 38, 38);
+}
+```
+
+
+### The React Application so far
+Your app so far should look like this:
+
+![React-PWA-image](/engineering-education/react-pwa/react-pwa1.png)
+
+![React-PWA-image2](/engineering-education/react-pwa/react-pwa2.png)
+
+
+### Adding functionality to make it a Progressive Web App
+In this part, we will now make our application to implement the Progressive Web App features. Some features of a progressive web app include:
```

Comment:
You can just say, 

`In this part, we will implement the Progressive Web App features. Some features of a progressive web app include:`